packages/nodes-from-anchor/src/v01/InstructionAccountNode.ts

```diff
@@ -22,6 +36,40 @@ export function instructionAccountNodeFromAnchorV01(idl: IdlV01InstructionAccoun
         defaultValue = publicKeyValueNode(idl.address, name);
     }
 
+    if (idl.pda) {
+        const [seeds, lookups] = idl.pda.seeds.reduce(
+            ([seeds, lookups], seed: IdlV01Seed) => {
+                const kind = seed.kind;
+
+                switch (kind) {
+                    case 'const':
+                        return [
+                            [...seeds, constantPdaSeedNodeFromBytes('base16', hex(new Uint8Array(seed.value)))],
+                            lookups,
+                        ];
+                    case 'account':
+                        return [
+                            [...seeds, variablePdaSeedNode(seed.path, publicKeyTypeNode())],
+                            [...lookups, pdaSeedValueNode(seed.path, accountValueNode(seed.path))],
+                        ];
+                    default:
+                        throw new KinobiError(KINOBI_ERROR__ANCHOR__SEED_KIND_UNIMPLEMENTED, {
+                            kind,
+                        });
```

Comment:
At this point, argument lookup is not supported because the account helper doesn’t have access to the arguments (though it can be updated to do so).

@lorisleiva, is Kinobi good to allow the type node for the argument field using variablePdaSeedNode(seed.path, typeNode(arg.type))?

Additionally, we are not handling the injection of seeds from deserialized accounts as we discussed, so treasury.mint support is not included, for example.

Also not handling injecting seed from deserialized account as we discussed so no `treasury.mint` support for example.

Reply:
Regarding the argument, yes you should be able to do something like this:

```ts
case 'argument':
    return [
        [...seeds, variablePdaSeedNode(seed.path, typeNodeFromAnchorV01(arg.type))],
        [...lookups, pdaSeedValueNode(seed.path, argumentValueNode(seed.path))],
    ];
```

You might just want to have an if statement first making sure `!seed.path.includes('.')`.

Reply:
Ok. Account for args then.

And for path based can it be kept as a variable but with nod pdaSeedValue? Kinobi keeps track of it has the correct type but caller is responsible for passing value?

Reply:
Interesting... You would need to add a new entry to the `extraArguments` array of the `instructionNode`. Then you can use this as an `argumentValueNode` in your PDA default value. That `extraArguments` array basically allows you to require more data to the end user when they create their instruction without including them in the instruction data.

That being said, for complex PDAs, it could end up adding lot's of attribute to the instruction input and make things worst so whilst we can do it, I'm not 100% sure that we should. Your call though. 😊

Reply:
Ok @lorisleiva though I could be wrong but I don't thinks its strategy used often but do think having the caller pass as the argument is the best way to handle this case. 

Also for args what are your thoughts and adding a link type field for argument types? 
 `typeArgumentLinkNode(seed.path)` instead of requiring the args to be passed to the `instructionAccountNodeFromAnchorV01` func.

Reply:
Hey, sorry I forgot to answer this one.

Currently all link nodes point to top-level nodes such as programs, instructions, etc. I think it may be confusing to introduce this sort of low-level link node. There may be a better way to avoid that redundancy though like introducing a different PDA value node altogether. I think we can revisit this in a subsequent PR though.